python/sparkdl/udf/keras_image_model.py

```diff
@@ -0,0 +1,90 @@
+#
+# Copyright 2017 Databricks, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import logging
+
+from ..graph.builder import GraphFunction, IsolatedSession
+from ..graph.pieces import buildSpImageConverter, buildFlattener
+from ..image.imageIO import imageSchema
+from ..utils import jvmapi as JVMAPI
+
+logger = logging.getLogger('sparkdl')
+
+def _serialize_and_reload_with(preprocessor):
+    """
+    Load a preprocessor function (image_file_path => image_tensor)
+    """
+    def udf_impl(spimg):
+        import numpy as np
+        from PIL import Image
+        from tempfile import NamedTemporaryFile
+        from sparkdl.image.imageIO import imageArrayToStruct, imageType
+        
+        pil_mode = imageType(spimg).pilMode        
+        img_shape = (spimg.width, spimg.height)
+        img = Image.frombytes(pil_mode, img_shape, bytes(spimg.data))
+        # Warning: must use lossless format to guarantee consistency
+        temp_fp = NamedTemporaryFile(suffix='.png')
+        img.save(temp_fp, 'PNG')
+        img_arr_reloaded = preprocessor(temp_fp.name)
+        assert isinstance(img_arr_reloaded, np.ndarray), \
+            "expect preprocessor to return a numpy array"        
+        img_arr_reloaded = img_arr_reloaded.astype(np.uint8)
+        return imageArrayToStruct(img_arr_reloaded)
+
+    return udf_impl
+
+def registerKerasImageUDF(udf_name, keras_model_or_file_path, preprocessor=None):
+    """
+    Create an UserDefinedFunction from a Keras model
+
+    registerKerasImageUDF(`udf_name`, "path/to/my/keras/model.h5", `preprocessor`)
+
+    :param udf_name: str, name of the UserDefinedFunction
+    :param keras_model_file_path: str, path to the HDF5 keras model file
+    :param preprocessor: function, optional, 
```

Comment:
you should explain what this is, and where it comes from. There should be an example above with inception, so that people understand the necessity. This is just a matter of copy/pasting one of the tests.